CorsixTH/Lua/world.lua

```diff
@@ -971,16 +971,30 @@ end
 
 --! Dedicated function to allow unpausing by pressing 'p' again
 function World:pauseOrUnpause()
+  if self:systemPauseStatus() then return end -- System pause takes precedence
   if not self:isCurrentSpeed("Pause") then
     self:setSpeed("Pause")
   elseif self.prev_speed then
     self:setSpeed(self.prev_speed)
   end
 end
 
+--! Sets whether the game should be forcefully paused
+--!param state (bool)
+function World:systemPause(state)
+  self.system_pause = state
+end
+
+--! Reports the system pause status
+--!return (bool) true is system pause is active, else false
+function World:systemPauseStatus()
+  return self.system_pause
```

Comment:
Oversight.. should probably be:
```suggestion
  return self.system_pause or false
```
Unless we want an afterload.

Reply:
And even with an afterload you should initialize its value for a new game.

Reply:
Yes my fear is I've been too slapdash on this. Oops.